packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts

```diff
@@ -269,19 +279,17 @@ export class AppSyncSwiftVisitor extends AppSyncModelVisitor {
 
   private getSwiftModelTypeName(field: CodeGenField) {
     if (this.isEnumType(field)) {
-      const name = this.getEnumName(field.type);
-      return field.isList ? `[${name}].self` : `${name}.self`;
+      return `${this.getEnumName(field.type)}.self`;
```

Comment:
confirm enum collections

Reply:
Could you

Reply:
this looks good, as long as it is an enum, it will be embedded inside `.embedded(type)` or `.embeddedCollection(of:)` when it is an array, without needing square brackets (as removed here)